pkg/operator/controllers/routefix/routefix.go

```diff
@@ -39,96 +35,25 @@ do
 done`
 )
 
-type RouteFix interface {
-	CreateOrUpdate(ctx context.Context) error
-}
-
-type routeFix struct {
-	log *logrus.Entry
-	env env.Interface
-
-	cli    kubernetes.Interface
-	seccli securityclient.Interface
-}
-
-func New(log *logrus.Entry, e env.Interface, cli kubernetes.Interface, seccli securityclient.Interface) RouteFix {
-	return &routeFix{
-		log: log,
-		env: e,
-
-		cli:    cli,
-		seccli: seccli,
-	}
-}
-
-func (i *routeFix) routefixImage() string {
-	return fmt.Sprintf(routefixImageFormat, i.env.ACRName())
-}
-
-func (i *routeFix) ensureNamespace(ctx context.Context, ns string) error {
-	_, err := i.cli.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        ns,
-			Annotations: map[string]string{projectv1.ProjectNodeSelector: ""},
-		},
-	}, metav1.CreateOptions{})
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_ns, err := i.cli.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		if _ns.Annotations == nil {
-			_ns.Annotations = map[string]string{}
-		}
-
-		if annotation, ok := _ns.Annotations[projectv1.ProjectNodeSelector]; annotation == "" && ok {
-			return nil
-		}
-
-		_ns.Annotations[projectv1.ProjectNodeSelector] = ""
-
-		_, err = i.cli.CoreV1().Namespaces().Update(ctx, _ns, metav1.UpdateOptions{})
-		return err
-	})
+func (r *RouteFixReconciler) routefixImage() string {
+	return fmt.Sprintf(routefixImageFormat, "arosvc")
```

Comment:
the name of the registry *must* be read out of the Cluster object ACRName field.